src/troubleshooting/content.md

```diff
@@ -0,0 +1,641 @@
+**TL;DR:** here's a diagram to help you debug your deployments in Kubernetes.
+
+![Flow chart to debug deployments in Kubernetes](diagram.png)
+
+When you wish to deploy an application in Kubernetes, you usually define three components:
+
+- a Deployment — which is a recipe for creating copies of your application called Pods
+- a Service — an internal load balancer that routes the traffic to Pods
+- an Ingress — a description of how the traffic should flow from outside the cluster to your Service.
+
+Here's a quick visual recap.
+
+```slideshow
+{
+  "description": "Kubernetes core concepts: Ingress, Service, Pod and Deployment.",
+  "slides": [
+    {
+      "image": "core-concepts-1.svg",
+      "description": "In Kubernetes your applications are exposed through two layers of load balancers: internal and external."
+    },
+    {
+      "image": "core-concepts-2.svg",
+      "description": "The internal load balancer is called Service, whereas the external one is called Ingress."
+    },
+    {
+      "image": "core-concepts-3.svg",
+      "description": "Pods are not deployed directly. Instead, the Deployment creates the Pods and whatches over them."
+    }
+  ]
+}
+```
+
+Assuming you wish to deploy a simple _Hello World_ application, the YAML for such application should look similar to this:
+
+```yaml|highlight=2,23,34|title=hello-world.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: app
+  labels:
+    track: canary
+spec:
+  selector:
+    matchLabels:
+      name: app
+  template:
+    metadata:
+      labels:
+        name: app
+    spec:
+      containers:
+      - name: app
```

Comment:
s/app/cont1/  <- people really don't know that this name is totally unused, so don't suggest that needs to match anything.

Reply:
Well, it's handy when you do a kubectl logs with multiple containers. But point taken and YAML amended.